### PR TITLE
[OpenPifPaf] ONNX Evaluation Pipeline

### PR DESCRIPTION
**Pull request by dbogunowicz**
## Feature Preview:

```bash
python3 -m openpifpaf.export_onnx --input-width 641 --input-height 641
deepsparse.pose_estimation.eval --model-path openpifpaf-resnet50.onnx  --dataset cocokp --image_size 641
```
Results in:
```bash
...
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] = 0.502
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets= 20 ] = 0.732
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets= 20 ] = 0.523
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = 0.429
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = 0.605
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] = 0.534
 Average Recall     (AR) @[ IoU=0.50      | area=   all | maxDets= 20 ] = 0.744
 Average Recall     (AR) @[ IoU=0.75      | area=   all | maxDets= 20 ] = 0.554
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = 0.457
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = 0.643
...
````
For more information please refer to the updated `README.md`

## Manual Testing
Compare the output above to the original openpifpaf eval:
```bash
python3 -m openpifpaf.eval --dataset cocokp --checkpoint shufflenetv2k16
```
Outputs:
```bash
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] = 0.501
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets= 20 ] = 0.732
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets= 20 ] = 0.515
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = 0.428
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = 0.604
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] = 0.532
 Average Recall     (AR) @[ IoU=0.50      | area=   all | maxDets= 20 ] = 0.742
 Average Recall     (AR) @[ IoU=0.75      | area=   all | maxDets= 20 ] = 0.549
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = 0.454
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = 0.643
```

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def _parse_requirements_file(file_path):
     "protobuf>=3.12.2,<=3.20.1",
     "click>=7.1.2,!=8.0.0",  # latest version < 8.0 + blocked version with reported bug
 ]
-_nm_deps = [f"{'sparsezoo' if is_release else 'sparsezoo-nightly'}~={version_base}"]
+_nm_deps = []#[f"{'sparsezoo' if is_release else 'sparsezoo-nightly'}~={version_base}"]
```
Comment (dbogunowicz) [inline setup.py:96]:
```suggestion
__nm_deps = [f"{'sparsezoo' if is_release else 'sparsezoo-nightly'}~={version_base}"]
```

File: setup.py [hunk 2]
```diff
@@ -93,7 +93,7 @@ def _parse_requirements_file(file_path):
     "protobuf>=3.12.2,<=3.20.1",
     "click>=7.1.2,!=8.0.0",  # latest version < 8.0 + blocked version with reported bug
 ]
-_nm_deps = [f"{'sparsezoo' if is_release else 'sparsezoo-nightly'}~={version_base}"]
+__nm_deps = [f"{'sparsezoo' if is_release else 'sparsezoo-nightly'}~={version_base}"]
```
Comment (dbogunowicz) [inline setup.py:96]:
```suggestion
_nm_deps = [f"{'sparsezoo' if is_release else 'sparsezoo-nightly'}~={version_base}"]
```

File: setup.py [hunk 3]
```diff
@@ -133,8 +133,10 @@ def _parse_requirements_file(file_path):
     "opencv-python<=4.6.0.66",
 ]
 _openpifpaf_integration_deps = [
-    "openpifpaf==0.13.6",
+    "openpifpaf==0.13.11",
```
Comment (dbogunowicz) [inline setup.py:136]: Would be great to also update this in SparseML. I am very confident that a minimal amount of work is needed.

**Review by bfineran — APPROVED** [review]

**Review by rahul-tuli — DISMISSED** [review]

**Review by bfineran — DISMISSED** [review]

**Review by bfineran — APPROVED** [review]

**Review by corey-nm — APPROVED** [review]
📈

**Pull request merged**